src/win/process.c

```diff
@@ -377,11 +377,13 @@ static WCHAR* search_path(const WCHAR *file,
   } else {
     dir_end = path;
 
-    /* The file is really only a name; look in cwd first, then scan path */
-    result = path_search_walk_ext(L"", 0,
-                                  file, file_len,
-                                  cwd, cwd_len,
-                                  name_has_ext);
+    if (getenv("NoDefaultCurrentDirectoryInExePath") == NULL || NeedCurrentDirectoryForExePath(cwd)) {
```

Comment:
```suggestion
    if (GetEnvironmentVariableW(L"NoDefaultCurrentDirectoryInExePath", NULL, 0) != 0) {
```
`getenv` is not threadsafe, so is best avoided

`NeedCurrentDirectoryForExePath(cwd)` is defined to return `TRUE`, so that check should not be included

Reply:
Actually, I guess the documentation for `NeedCurrentDirectoryForExePathW` says that the name of the environment variable is in the registry and might change, so it would be better to call this function instead:
```suggestion
    if (NeedCurrentDirectoryForExePathW(L"")) {
```

Reply:
Updated